### ISSUE TITLE
Data Channel connection wouldn't establish with logs from DTLS

### ISSUE THREAD

**Issue by enobufs** [pions/dtls#42]
### Your environment.
  - Version: *Release or SHA*: 44ae5 (dtls v1.2.1, webrtc@v1.2.1-0.20190218202731-31fa14cd0d01)
  - Browser: *include version*: n/a
  - Other Information - See below

### What did you do?
Running data channel connection tests over the Internet, noticed that data channel connection wouldn't establish (never completes) close to 60% of chances (it varies depending on the location of endpoints) 

### What did you expect?
Data Channel connection is successfully established.

### What happened?
Data channel connection establishment wouldn't complete with high occurrence ratio, ~60%. When that happens, saw following log messages would show up always. 

```
handleIncoming: old epoch, dropping packet
handleIncoming: old epoch, dropping packet
handleIncoming: old epoch, dropping packet
handleIncoming: old epoch, dropping packet
CipherSuite has not been initialized, unable to decrypt
```

The above repeats forever until the application gives and closes the peer connection.

Other information:
* This never happens on the local environment
* We are using only one data channel for now (no media used)

**Comment by Sean-Der** [comment]
Hey @enobufs 

This just looks like poor lossy connection handling, this should be easy enough  to fix! I probably made a poor assumption somewhere :(

Are you using pion on both sides? Do you only get error messages on one side? The side that you see error messages is it the offerer or answerer?

**Comment by enobufs** [comment]
Yes, using pion on both ends. Those logs were from offerer (connecting) side. (I don't have the server side log...)

**Comment by enobufs** [comment]
The rev at b388d3f seems to work very well!

First, I was able to reproduce the problem (without the fix) with Network Link Conditioner (mac) with the following settings:
```
Downlink: 4% loss, 200ms delay, 1Mbps bandwidth
Downlink: 4% loss, 200ms delay, 256Kbps bandwidth
```

With this configuration, I saw the CiperSuite/old epoch logs at about 50 percent of chances.

Then I used an experimental branch of pions/webrtc, I created, which includes dtls@b388d3f and sctp@master (t1-init/cookie).  I did not see those logs from dtls at all, and data channel connected every time - huge improvement!
